src/main/java/io/openliberty/tools/common/plugins/util/ServerFeatureUtil.java

```diff
@@ -322,9 +329,39 @@ private static boolean isURL(String url) {
                 } else {
                     result.addAll(features);
                 }
-            }  
+            }
         }
         return result;
     }
+
+    private Properties getBootstrapProperties(File bootstrapProperties) {
+        Properties prop = new Properties();
+        if (bootstrapProperties != null && bootstrapProperties.exists()) {
+            try {
+                prop.load(new FileInputStream(bootstrapProperties));
+            } catch (IOException e) {
+                warn("The bootstrap.properties file " + bootstrapProperties + " could not be loaded. Skipping the bootstrap.properties file.");
+                debug(e);
+            }
+        }
+        return prop;
+    }
+
+    private String evaluateExpression(Properties properties, String expression) {
+        String value = expression;
+        if (expression != null) {
+            Pattern p = Pattern.compile("\\$\\{([^\\}]*)\\}");
+            Matcher m = p.matcher(expression);
+            StringBuffer sb = new StringBuffer();
+            while (m.find()) {
+                String variable = m.group(1);
+                String propertyValue = properties.getProperty(variable, "\\$\\{" + variable + "\\}");
+                m.appendReplacement(sb, propertyValue);
```

Comment:
Does this guard against recursive references?

Reply:
Not entirely sure what you mean.  If you means a variable that references another variable then no it does not support that.  Similar to my other comment this supports a minimal use case of properties defined in bootstrap properties similar to what was done in `HttpPortUtil`.  

Anything more would be a lot more work and honestly I find it surprising that there isn't already a comprehensive API for reading in a servers configuration.  Liberty has to already have something built into it that handles all this so why that isn't just broken into a separate library that can be shared between the maven plugin and app server runtime is beyond me.